### PR TITLE
feat(way): parse osmApiWay to osmWay

### LINKED ISSUE #17: Parse osm-api way response to OsmWay interface
The way element's "nodes" key in the response from the osm-api is an array of node ids.
The `OsmWay` interface expects an array of `OsmNodes`.
Create a parsing logic between the way in osm-api response and the `OsmWay` interface.

### PR DESCRIPTION
**Pull request by melancholiai**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix        | ✖                                                                         |
| New feature     | ✔                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation  | ✖                                                                         |
| Tests added     | ✔                                                                        |
| Chore            | ✔                                                                       |

Closes #17

### CODE REVIEW

File: commitlint.config.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module.exports = {
   extends: ["@commitlint/config-conventional"],
   rules: {
-    "scope-enum": [2, "always", ["node", "way", "relation", "changeset", "osmChange", "dependencies", "configurations"]],
+    "scope-enum": [2, "always", ["util", "node", "way", "relation", "changeset", "osmChange", "deps", "configurations"]],
```
Comment (syncush) [inline commitlint.config.js]: the feature is on way (`parseOsmWayApi()`) therefore the commit message does not fit.

File: package.json [hunk 2]
```diff
@@ -4,18 +4,20 @@
...
     "release": "standard-version",
+    "test": "jest --config=./tests/configurations/jest.config.js",
     "build": "tsc --project tsbuildconfig.json"
   },
   "repository": {
     "type": "git",
     "url": "git+https://github.com/MapColonies/node-osm-elements.git"
   },
-  "author": "galta95",
+  "author": "MapColonies",
```
Comment (syncush) [inline package.json:20]: good catch

File: src/util/util.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { BaseElement, OsmNode, OsmWay, OsmApiWay, Types } from '..';
```
Comment (syncush) [inline src/util/util.ts]: now that we have a util function I would move all the interfaces to interfaces folder with all the proper interfaces and a proper `index.ts`, then import it nicely in this `util.ts` file, our `index.ts` should be minimal at this point we have BL in our package.

File: src/index.ts [hunk 4]
```diff
@@ -9,24 +14,30 @@ interface BaseElement {
...
-  type: 'way';
+  type: Types.WAY;
   nodes: OsmNode[];
 }
 
+interface OsmApiWay extends BaseElement {
+  type: Types.WAY;
+  nodes: number[];
+}
+
```
Comment (syncush) [inline src/index.ts]: is it for only full api call or even not full ?

File: src/index.ts [hunk 5]
```diff
@@ -43,4 +54,4 @@ interface OsmChange {
   delete?: BaseElement[];
 }
 
-export { BaseElement, OsmNode, OsmWay, OsmRelation, Changeset, OsmChange, Member };
+export { BaseElement, OsmNode, OsmWay, OsmApiWay, OsmRelation, Changeset, OsmChange, Member, Types };
```
Comment (syncush) [inline src/index.ts]: exporting Types as Types is a bit problematic IMO, we should name it ElementsType or something like that, Types is too broad

File: src/index.ts [hunk 6]
```diff
@@ -9,24 +14,30 @@ interface BaseElement {
...
 };
 
 interface OsmRelation extends BaseElement {
-  type: 'relation';
+  type: Types.RELATION;
   members?: Member[];
 }
+
 interface Changeset {
   type: 'changeset';
```
Comment (syncush) [inline src/index.ts]: why have you decided not to include `changeset` in the `Types` enum? same for `osmchange`

**Review by syncush — CHANGES_REQUESTED** [review]

File: src/util/util.ts (hunk 3 above)
Comment (melancholiai) [reply src/util/util.ts]: done.

File: package.json (hunk 2 above)
Comment (melancholiai) [reply package.json:20]: eye surgery can be declared successful

File: commitlint.config.js (hunk 1 above)
Comment (melancholiai) [reply commitlint.config.js]: removed util, should I only rename the pr or also all commits with util in them?

File: src/index.ts (hunk 4 above)
Comment (melancholiai) [reply src/index.ts]: it's for all apis. in the full endpoint along with the way you get it's nodes

File: src/index.ts (hunk 5 above)
Comment (melancholiai) [reply src/index.ts]: will name them all under `OsmTypes`

File: src/index.ts (hunk 6 above)
Comment (melancholiai) [reply src/index.ts]: if I add them, it shouldn't be renamed as `ElementsType` (as mentioned in the comment above).
named them as `OsmTypes`

**Review by syncush — APPROVED** [review]

**Pull request merged**